x-pack/solutions/security/plugins/elastic_assistant/server/routes/attack_discovery/schedules/find.test.ts

```diff
@@ -0,0 +1,133 @@
+import {
+  getInternalFindAttackDiscoverySchedulesMock,
+  getInternalAttackDiscoveryScheduleMock,
+} from '../../../__mocks__/attack_discovery_schedules.mock';
+import { AttackDiscoveryScheduleDataClient } from '../../../lib/attack_discovery/schedules/data_client';
+
+const { clients, context } = requestContextMock.createTools();
+const server: ReturnType<typeof serverMock.create> = serverMock.create();
+clients.core.elasticsearch.client = elasticsearchServiceMock.createScopedClusterClient();
+
+const findAttackDiscoverySchedule = jest.fn();
+const mockSchedulingDataClient = {
+  findSchedules: findAttackDiscoverySchedule,
+  getSchedule: jest.fn(),
+  createSchedule: jest.fn(),
+  updateSchedule: jest.fn(),
+  deleteSchedule: jest.fn(),
+  enableSchedule: jest.fn(),
+  disableSchedule: jest.fn(),
+} as unknown as AttackDiscoveryScheduleDataClient;
+const mockApiConfig = {
+  connectorId: 'connector-id',
+  actionTypeId: '.bedrock',
+  model: 'model',
+  provider: OpenAiProviderType.OpenAi,
+};
+const basicAttackDiscoveryScheduleMock = {
+  name: 'Test Schedule',
+  schedule: {
+    interval: '100m',
+  },
+  params: {
+    alertsIndexPattern: '.alerts-security.alerts-default',
+    apiConfig: mockApiConfig,
+    end: 'now',
+    size: 25,
+    start: 'now-24h',
+  },
+  enabled: true,
+};
+
+describe('findAttackDiscoverySchedulesRoute', () => {
+  beforeEach(() => {
+    jest.clearAllMocks();
+    context.elasticAssistant.getAttackDiscoverySchedulingDataClient.mockResolvedValue(
+      mockSchedulingDataClient
+    );
+    context.elasticAssistant.getRegisteredFeatures.mockReturnValue({
+      ...defaultAssistantFeatures,
+      assistantAttackDiscoverySchedulingEnabled: true,
+    });
+    findAttackDiscoverySchedulesRoute(server.router);
+    findAttackDiscoverySchedule.mockResolvedValue(
+      getInternalFindAttackDiscoverySchedulesMock([
+        getInternalAttackDiscoveryScheduleMock(basicAttackDiscoveryScheduleMock),
+      ])
+    );
+  });
+
+  it('should handle successful request', async () => {
+    const response = await server.inject(
+      findAttackDiscoverySchedulesRequest(),
+      requestContextMock.convertContext(context)
+    );
+    expect(response.status).toEqual(200);
+    expect(response.body).toEqual({
+      page: 1,
+      perPage: 20,
+      total: 1,
+      data: [expect.objectContaining(basicAttackDiscoveryScheduleMock)],
+    });
+  });
+
+  it('should handle missing data client', async () => {
+    context.elasticAssistant.getAttackDiscoverySchedulingDataClient.mockResolvedValue(null);
+    const response = await server.inject(
+      findAttackDiscoverySchedulesRequest(),
+      requestContextMock.convertContext(context)
+    );
+
+    expect(response.status).toEqual(500);
+    expect(response.body).toEqual({
+      message: 'Attack discovery data client not initialized',
+      status_code: 500,
+    });
+  });
+
+  it('should handle `dataClient.createSchedule` error', async () => {
```

Comment:
nit: Consider `createSchedule` -> `findSchedules`

Reply:
done in https://github.com/elastic/kibana/pull/216656/commits/89adba597b631e0c7d4a055927a7fb2543cc61b7